### PR TITLE
refactor private amop implement

### PR DESCRIPTION
**Pull request by bxq2011hust**

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- examples/amop_auth/unicast_pub/publisher.go  1
         

Complexity decreasing per file
==============================
+ conn/channel.go  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6149296&bid=20096258)

**Review by ywy2090 — DISMISSED** [review]

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- examples/amop_auth/unicast_pub/publisher.go  1
- examples/amop/broadcast_pub/publisher.go  5
         

Complexity decreasing per file
==============================
+ conn/channel.go  -3
         

Clones added
============
- examples/amop/broadcast_pub/publisher.go  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6149296&bid=20096258)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- examples/amop/broadcast_pub/publisher.go  5
- examples/amop_auth/unicast_pub/publisher.go  1
         

Complexity decreasing per file
==============================
+ conn/channel.go  -3
         

Clones added
============
- examples/amop/broadcast_pub/publisher.go  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6149296&bid=20096258)

**Review by ywy2090 — APPROVED** [review]

File: conn/channel.go [hunk 1]
```diff
@@ -60,9 +60,7 @@ type channelSession struct {
 	// receiptsMutex sync.Mutex
 	receiptResponses    map[string]*channelResponse
 	topicMu             sync.RWMutex
-	topicHandlers       map[string]func([]byte)
-	messageMu           sync.RWMutex
-	messageHandlers     map[string]func([]byte)
+	topicHandlers       map[string]func([]byte, *[]byte)
```
Comment (sulenn) [inline conn/channel.go:63]: why to add *[]byte?

File: conn/channel.go [hunk 2]
```diff
@@ -898,23 +853,11 @@ func (hc *channelSession) processTopicMessage(msg *channelMessage) {
...
-		jsonBytes, err := json.Marshal(jsonStruct)
-		if err != nil {
-			// fmt.Printf("json marshal failed, err: %v\n", jsonBytes)
-			return
-		}
-		go handler(jsonBytes)
-		return
+	responseData := new([]byte)
+	if ok {
+		handler(topic.data, responseData)
```
Comment (sulenn) [inline conn/channel.go:858]: the parameter of *[]byte is only used to process random number signatures? How to use by user?

**Review by sulenn — COMMENTED** [review]
nice job!

**Pull request merged**